src/Datapackages/BaseDatapackage.php

```diff
@@ -183,20 +183,27 @@ public function save($zip_filename)
     {
         Package::isZipPresent();
         $zip = new ZipArchive();
+
+        $packageCopy = $this->copy();
+
         $base = tempnam(sys_get_temp_dir(), 'datapackage-zip-');
         $files = [
             'datapackage.json' => $base.'datapackage.json',
         ];
         $ri = 0;
-        foreach ($this as $resource) {
+        foreach ($packageCopy->resources() as $resource) {
```

Comment:
@jonrmitchell hi. Since this class extends /Iterator I think you can iterate directly over $packageCopy here without having to request all resources. The cursor will start at 0 and may be more optimal for large Packages?

Reply:
Oh, that makes sense and I'd be fine with that.